cmd/derper/cert.go

```diff
@@ -76,6 +98,18 @@ func NewManualCertManager(certdir, hostname string) (certProvider, error) {
 	if err := x509Cert.VerifyHostname(hostname); err != nil {
 		return nil, fmt.Errorf("cert invalid for hostname %q: %w", hostname, err)
 	}
+	if _, err := netip.ParseAddr(hostname); err == nil {
+		// If the hostname is an IP address, print out information on how to
+		// confgure this in the derpmap.
+		dn := &tailcfg.DERPNode{
+			Name:     "custom",
+			RegionID: 900,
+			HostName: hostname,
+			CertName: fmt.Sprintf("sha256-raw:%-02x", sha256.Sum256(x509Cert.Raw)),
+		}
+		dnJSON, _ := json.Marshal(dn)
+		log.Printf("Using self-signed certificate for IP address %q. Configure it in DERPMap using:\n  %s", hostname, dnJSON)
```

Comment:
Link to [KB](https://tailscale.com/kb/1232/derp-servers#customize-your-derp-map)?

Reply:
We never link directly to the kb from the client code and only go via a `tailscale.com/s/*` short URL that we can update the URL mapping later. We could do that, but I'm not sure it's worthwhile: anybody using this is already using a custom DERP map, so they know how to do it. And I'd bet 90% of the people interested in this feature are using Headscale instead of Tailscale which has different instructions.

So I'm not opposed, but I question the value of hand holding too much here.